app/src/main/java/org/mozilla/rocket/theme/ThemeManager.java

```diff
@@ -0,0 +1,136 @@
+package org.mozilla.rocket.theme;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.res.Resources;
+import android.preference.PreferenceManager;
+
+import org.mozilla.focus.R;
+
+import java.util.HashSet;
+
+public class ThemeManager {
+
+    private static final int ONBOARDING_VERSION = 1;
+
+    private static final String PREF_KEY_STRING_CURRENT_THEME = "pref_key_string_current_theme";
+    private static final String PREF_KEY_INT_ONBOARDING_VERSION = "pref_key_int_onboarding_version";
+
+    public interface Themeable {
+        void onThemeChanged();
```

Comment:
onThemeChanged(int themeId) could be a better option IMO.

Reply:
I'm okay if we add the ```themeId``` as a param. Just curious that if we should let the Themeable know what is the changed theme? Or we only notify and they just set the value from latest themed resources, no matter what the ```themeId``` is?